### DISCUSSION TITLE
Removing `stream.pipe(req)` functionality

### DISCUSSION THREAD

**Discussion by jimmywarting** [ladjs/lad#443] [General]
If the intent is to use fetch / Request instead
...then you "can't" really pipe a stream to request. you instead use a stream as a body. 
so it's kind of like piping but yet it isn't in the terminology of NodeJS streams.

so with fetch you would actually have to do something like 
```js
fetch(url, {
  method: 'POST',
  body: new ReadableStream(),
  duplex: 'full'
})
```

The fetch spec do also support async iterator that yield uint8arrays (so that it can accept both web & node streams)
so undici, web and NodeJS do all support generators that yields uint8array as a body...
```js
function * gen() {
  yield new Uint8Array(0)
}

fetch(url, {
  method: 'POST',
  body: gen(),
  duplex: 'full'
})
```

So with that in mind i would suggest that pipe & write gets removed.
it should be replaced with a `.send()` method instead which acts more or less like a "body" -setter

So i would suggest that this example code ([here](https://ladjs.github.io/superagent/#piping-data)) becomes more like:
```diff
  import request from 'superagent'
  import fs from 'node:fs'

  const stream = fs.createReadStream('path/to/my.json')
- const req = request.post('/somewhere')
- req.type('json')
- stream.pipe(req)

+ request.post('/somewhere')
+   .type('json')
+   .send(stream)
```

neither `XMLHttpRequest` and `fetch` do not have this `stream.pipe(req)` equally functionality. and it's considerable much harder to support it with some kind of own hacky `RequestBase.prototype.pipe|write` functionality then it's to just simple handle it the proper way in a `.send()` method.

Making superagents `Request` extend `node:streams` makes it considerable much harder to extend `BaseRequest` cuz it instead depends on mixins to add functionality onto classes

So it makes it hard to refactor the prototype based codespace into classes when i can't use `class X extends Y`.

The web it dose not even inherit stream and you can't pipe stuff at all in the web version of superagent. so removing pipe, write and inheriting streams makes more sense.

inherit (or extending) `node:streams` means it would have to polyfill a bunch of things for non-NodeJS env and including buffer, EventEmitter, text_decoder and a bunch of things that are not a good fit for eg web or Deno.

**Comment by jimmywarting** [comment]
ping @titanism

**Comment by titanism** [comment]
def down to mirror undici/fetch behavior more (hopefully using undici under the hood server-side instead of xmlhttprequest and fetch client-side) and do a semver major release

**Comment by titanism** [comment]
@jimmywarting feel free to take the reigns with this effort and use your best judgment, love the thought and mental cycles you're putting in here, very very stoked to see what you come up with

**Reply by jimmywarting** [reply]
Grate that we are on the same page so I know what I can help out and contribute to
